vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt

```diff
@@ -162,21 +168,34 @@ class CreateRoomViewModel @AssistedInject constructor(@Assisted private val init
     private fun setTopic(action: CreateRoomAction.SetTopic) = setState { copy(roomTopic = action.topic) }
 
     private fun setIsPublic(action: CreateRoomAction.SetIsPublic) = setState {
+        val roomAccessRules = if (action.restricted) RoomAccessRules.RESTRICTED else RoomAccessRules.UNRESTRICTED
+        val roomVisibilityType = when {
+            action.isPublic   -> CreateRoomViewState.RoomVisibilityType.Public("")
+            action.restricted -> CreateRoomViewState.RoomVisibilityType.Private
+            else              -> CreateRoomViewState.RoomVisibilityType.External
+        }
         if (action.isPublic) {
+            val userHSDomain = TchapUtils.getHomeServerDisplayNameFromMXIdentifier(session.myUserId)
+            val isAgentServerDomain = userHSDomain.equals(AGENT_SERVER_DOMAIN, ignoreCase = true)
             copy(
-                    roomVisibilityType = CreateRoomViewState.RoomVisibilityType.Public(""),
+                    roomVisibilityType = roomVisibilityType,
+                    roomAccessRules = roomAccessRules,
                     // Reset any error in the form about alias
                     asyncCreateRoomRequest = Uninitialized,
-                    isEncrypted = false
+                    isEncrypted = false,
+                    // Public rooms are not federated by default except for agent server domain
+                    disableFederation = !isAgentServerDomain,
+                    isFederationSettingAvailable = !isAgentServerDomain
+
             )
         } else {
             copy(
-                    roomVisibilityType = if (action.restricted) {
-                        CreateRoomViewState.RoomVisibilityType.Private
-                    } else {
-                        CreateRoomViewState.RoomVisibilityType.External
-                    },
-                    isEncrypted = adminE2EByDefault
+                    roomVisibilityType = roomVisibilityType,
+                    roomAccessRules = roomAccessRules,
+                    isEncrypted = adminE2EByDefault,
+                    // Private rooms are all federated
+                    disableFederation = false,
+                    isFederationSettingAvailable = true
```

Comment:
I would set false this value `isFederationSettingAvailable` because it is not used in this case